### PR TITLE
Added module for Secure Web Proxy

### PR DESCRIPTION
**Pull request by rosmo**
Also fixes two issues with tests with newer provider.

### CODE REVIEW

**Comment by rosmo** [comment]
Feel free to review at this point though.

File: modules/net-vpc-swp/main.tf [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  provider              = google-beta
+  project               = var.project_id
+  name                  = var.name
+  location              = var.region
+  ca_pool               = var.tls_inspection_config.ca_pool
+  exclude_public_ca_set = var.tls_inspection_config.exclude_public_ca_set
+}
+
+resource "google_network_security_gateway_security_policy_rule" "secure_tag_rules" {
+  for_each                = try(var.policy_rules.secure_tags, {})
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:41]: this try is not needed

File: modules/net-vpc-swp/main.tf [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  provider    = google-beta
+  project     = var.project_id
+  name        = each.key
+  location    = var.region
+  description = "Managed by Terraform."
+  values      = each.value.values
+}
+
+resource "google_network_security_gateway_security_policy_rule" "url_list_rules" {
+  for_each                = try(var.policy_rules.url_lists, {})
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:68]: this try is not needed

File: modules/net-vpc-swp/main.tf [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+${each.value.url_list}
+%{~endif~}') %{if each.value.session_matcher != null~} && (${each.value.session_matcher})%{endif~}
+EOT
+  application_matcher     = each.value.application_matcher
+  tls_inspection_enabled  = each.value.tls_inspection_enabled
+  basic_profile           = each.value.action
+}
+
+resource "google_network_security_gateway_security_policy_rule" "custom_rules" {
+  for_each                = try(var.policy_rules.custom, {})
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:89]: this try is not needed

File: modules/net-vpc-swp/main.tf [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  basic_profile           = each.value.action
+}
+
+resource "google_network_services_gateway" "gateway" {
+  provider                             = google-beta
+  project                              = var.project_id
+  name                                 = var.name
+  location                             = var.region
+  labels                               = var.labels
+  addresses                            = var.addresses != null ? var.addresses : []
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:109]: would it make sense to make addresses non-nullalble with `[]` as default?

File: modules/net-vpc-swp/main.tf [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  type                                 = "SECURE_WEB_GATEWAY"
+  ports                                = var.ports
+  scope                                = var.scope != null ? var.scope : ""
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:112]: coalesce(var.scope, "")

File: modules/net-vpc-swp/variables.tf [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  validation {
+    condition = (
+      length(concat(
+        [for k, v in try(var.policy_rules.secure_tags, {}) : v.priority],
+        [for k, v in try(var.policy_rules.url_lists, {}) : v.priority],
+      [for k, v in try(var.policy_rules.custom, {}) : v.priority])) ==
+      length(distinct(concat(
+        [for k, v in try(var.policy_rules.secure_tags, {}) : v.priority],
+        [for k, v in try(var.policy_rules.url_lists, {}) : v.priority],
+      [for k, v in try(var.policy_rules.custom, {}) : v.priority])))
```
Comment (juliocc) [inline modules/net-vpc-swp/variables.tf:95]: none of these trys are needed either

File: modules/net-vpc-swp/main.tf (hunk 5 above)
Comment (rosmo) [reply modules/net-vpc-swp/main.tf:112]: Can't do:
```
modules/net-vpc-swp/main.tf line 120, in resource "google_network_services_gateway" "gateway":
 120:   scope                                = coalesce(var.scope, "")
    ├────────────────
    │ while calling coalesce(vals...)
    │ var.scope is null

Call to function "coalesce" failed: no non-null, non-empty-string arguments.
```

Comment (juliocc) [reply modules/net-vpc-swp/main.tf:112]: ignore then ;)

File: modules/net-vpc-swp/main.tf [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+resource "google_network_security_gateway_security_policy_rule" "url_list_rules" {
+  for_each                = var.policy_rules.url_lists
+  provider                = google-beta
+  project                 = var.project_id
+  name                    = each.key
+  location                = var.region
+  gateway_security_policy = google_network_security_gateway_security_policy.policy.name
+  enabled                 = each.value.enabled
+  priority                = each.value.priority
+  session_matcher         = <<EOT
```
Comment (juliocc) [inline modules/net-vpc-swp/main.tf:76]: Can you indent this heredoc like you did with the other one (`<<-EOT`)

Comment (rosmo) [reply modules/net-vpc-swp/main.tf:76]: I tried but it feels like a Terraform bug that it doesn't strip all the whitespace correctly with the indented version and it results in whitespaces that mess with the tests.

Comment (rosmo) [reply modules/net-vpc-swp/main.tf:76]: Looks like this:
```
E             assert "inUrlList(ho...ccount.com'))" == "inUrlList(ho...ccount.com'))"
E               - inUrlList(host(), 'projects/my-project/locations/europe-west4/urlLists/my-url-list') && (source.matchServiceAccount('my-sa@my-project.iam.gserviceaccount.com'))
E               + inUrlList(host(), '    projects/my-project/locations/europe-west4/urlLists/my-url-list
E               + ')&& (source.matchServiceAccount('my-sa@my-project.iam.gserviceaccount.com'))
```

**Comment by rosmo** [comment]
@juliocc should be ok now!

**Review by juliocc — APPROVED** [review]

**Pull request merged**